### ISSUE TITLE
Blocks: differentiate premium blocks from free ones when customer doesn't have paid plan

### ISSUE THREAD

**Issue by thomasguillot** [Automattic/jetpack#11875]
Update 9th August 2019: premium blocks now discoverable for free customers in the block picker, I've changed this issue to be more specific about the presentation of these blocks in the block picker since that's what these designs are concentrating anyway. - @simison

---

As stated in the documentation:

> The feature is available to Premium and Business plan subscribers on WordPress.com and on Premium and Professional plans for Jetpack sites running version 5.2 or higher.

Which makes this block completely hidden for other users.

Here, I suggest:

1. Show the block button but disabled
2. Attach a tooltip to it (different content depending on Jetpack or WPcom user)
3. Link the tooltip to an upgrade screen

![screenshot 2018-11-29 at 11 09 55](https://user-images.githubusercontent.com/177929/49218568-a7431400-f3c8-11e8-9ddd-5d044ebf2ff5.png)

And use this pattern for all future blocks requiring a plan.

**Comment by simison** [comment]
A couple more ideas:

1. Separate categories were explored in https://github.com/Automattic/jetpack/pull/12431
    <img width="440" alt="Screenshot 2019-05-21 at 13 59 20" src="https://user-images.githubusercontent.com/275961/58098821-3db2fe00-7bd2-11e9-8534-97a03ddeae2a.png">

2. Show different icon color
    <img width="479" alt="Screenshot 2019-08-09 at 09 32 15" src="https://user-images.githubusercontent.com/87168/62759098-05acb000-ba89-11e9-80fb-ee1eb3fe65f6.png">

3. Show coloured border
    <img width="521" alt="Screenshot 2019-08-09 at 09 32 33" src="https://user-images.githubusercontent.com/87168/62759097-05141980-ba89-11e9-948c-e74e2695ccce.png">

4. Add "Premium" suffix to block name
    <img width="516" alt="Screenshot 2019-08-09 at 09 33 51" src="https://user-images.githubusercontent.com/87168/62759143-2f65d700-ba89-11e9-8ba6-11155d7b4e75.png">


Note that this menu also would need similar mechanism:

<img width="338" alt="image" src="https://user-images.githubusercontent.com/87168/62759330-a602d480-ba89-11e9-958b-bdd0f73e037d.png">

**Comment by simison** [comment]
Idea via @mtias:

> Don’t mind the presentation, but using the block description card for this could be interesting
> ![image](https://user-images.githubusercontent.com/87168/74191997-55994900-4c5d-11ea-9087-bcf79ed5be2e.png)

(PR for help panel in core https://github.com/WordPress/gutenberg/pull/16813)

**Comment by obenland** [comment]
More ideas via @lessbloat:

![image](https://p111.p2.n0.cdn.getcloudapp.com/items/jkunzRZK/paid-block-vs-free.png?v=2ad18a49048125f471cf4afdced7f71c)

**Comment by WunderBart** [comment]
I really like the idea of the badge icon from https://github.com/Automattic/jetpack/issues/11875#issuecomment-584377505. To address the currency issue (cc @davemart-in), I think that instead of using the dollar sign we could use e.g. [a star icon](https://icons-for-free.com/best+bookmark+premium+rating+select+star+icon-1320168257340660520/) as an indication of a premium entity. I also think that explaining what that icon means as suggested here https://github.com/Automattic/jetpack/issues/11875#issuecomment-584366823 would be a good idea. The premium icon would also be convenient as an indication in the `/` triggered list (re: https://github.com/Automattic/jetpack/issues/11875#issuecomment-519799019).

It could look more-less like this:

![jetpack paid blocks selector design](https://user-images.githubusercontent.com/1451471/74237536-135f1e80-4cd4-11ea-92c1-d028f013a09c.png)

**Comment by davemart-in** [comment]
I really like that @WunderBart. The only minor change that I would make would be to make the orange border around the paid plan description in the block preview the same color green as the icon. Nice work! Let's do this! Let me know what you need from me to make this happen.

**Comment by obenland** [comment]
@WunderBart @jeryj Based on the latest comments on paYJgx-rs-p2#comment-659, let's also separate these premium blocks out into their own category.

**Comment by WunderBart** [comment]
Thanks for the update, @obenland. To summarize the direction we've decided to take - we're going to:
- Add the `paid` label to block title if a current paid plan doesn't cover it (already started in https://github.com/Automattic/jetpack/pull/14739) [paYJgx-rs-p2#comment-656],
- Separate premium blocks out into their own category (probably will happen in a separate PR) [paYJgx-rs-p2#comment-659].

cc @davemart-in, @jeryj.

**Comment by jeherve** [comment]
This has now been built, so this issue can be closed. Thanks everyone!
